### DISCUSSION TITLE
Next.js 12.1.0 - CSP not working after exporting HTMLs

### DISCUSSION THREAD

**Discussion by sm1990** [vercel/next.js#48184] [Help]
### Summary

Hi,

I have a next.js site which I deploy via GitHub pages. For that I've added the following CSP in `next.config.js` file.
```
/** @type {import('next').NextConfig} */

const redirectBase = process.env.NEXT_PUBLIC_BASE_PATH
  ? `${process.env.NEXT_PUBLIC_BASE_PATH}/`
  : "";

const securityHeaders = [
  {
    key: 'X-Frame-Options',
    value: 'SAMEORIGIN',
  },
  {
    key: 'Content-Security-Policy',
    value: "frame-ancestors 'none';"
  }
]


const nextConfig = {
  reactStrictMode: true,
  basePath: process.env.NEXT_PUBLIC_BASE_PATH || "",
  assetPrefix: process.env.NEXT_PUBLIC_BASE_PATH || "",
  images: {
    loader: "custom",
    imageSizes: [16, 32, 48, 64, 96, 128, 256, 384],
    deviceSizes: [640, 750, 828, 1080, 1200, 1920, 2048, 3840],
    nextImageExportOptimizer: {
      imageFolderPath: "public/images",
      exportFolderPath: "out",
      quality: 75,
    },
  },
  async headers() {
    return [
      {
        // Apply these headers to all routes in your application.
        source: '/:path*',
        headers: securityHeaders,
      },
    ]
  },
  trailingSlash: true,
  resolve: [{ path: false, fs: false }],
  experimental: {
    scrollRestoration: true,
  }
};

module.exports = nextConfig;
```
I build the site using, `npm run build` and export the HTMLs using `npm run export`. However when I check the headers after deploying the site I noticed that the CSP rules are missing. Any idea why that is happening and hoe I can fix it?

Thanks in advance.

**Comment by philwolstenholme** [comment]
HTTP headers are set by the server, not the HTML files themselves. When you export a Next site you are only exporting the HTML files, and GitHub Pages becomes your server.

I don't think GitHub Pages supports setting HTTP headers. But you could try a competitor that does, like [Netlify](https://docs.netlify.com/routing/headers/) or [Cloudflare Pages](https://developers.cloudflare.com/pages/platform/headers/).
